File: le1/plp/expressions1/expression/ValorBooleano.java
package le1.plp.expressions1.expression;

import le1.plp.expressions1.util.Tipo;

/**
 * Este valor primitivo encapsula um valor booleano.
 */
public class ValorBooleano extends ValorConcreto<Boolean> {

	
	/**
	 * Cria um objeto encapsulando o valor booleano fornecido.
	 */
	public ValorBooleano(boolean valor) {
		super(valor);
	}

	
	/**
	 * Retorna os tipos possiveis desta expressao.
	 *
	 * @return os tipos possiveis desta expressao.
	 */
	public Tipo getTipo() {
		return Tipo.TIPO_BOOLEANO;
	}


}

File: Le1 - Python/le1/expression/valor_booleano.py
from __future__ import annotations
from le1.expression.valor import Valor
from le1.expression.valor_concreto import ValorConcreto
from abc import ABC
from typing import Generic, TypeVar

from le1.expression.expressao import Expressao
from le1.util.tipo import Tipo

T = TypeVar("T")

class ValorBooleano(ValorConcreto[bool]):
    def __init__(self, valor: bool) -> None:
        super().__init__(valor)

    def getTipo(self) -> Tipo:
        return Tipo.TIPO_BOOLEANO

    def __str__(self) -> str:
        return str(self._valor).lower()

